pkg/csi_driver/controller.go

```diff
@@ -830,7 +827,7 @@ func (s *controllerServer) DeleteSnapshot(ctx context.Context, req *csi.DeleteSn
 
 	if !isBackup {
 		glog.Errorf("Deletion of volume snapshot type %q not supported", id)
-		return nil, status.Error(codes.Internal, "Deletion of volume snapshot type not supported")
+		return nil, status.Error(codes.InvalidArgument, "Deletion of volume snapshot type not supported")
```

Comment:
nit: maybe say something like "only deletion of Backup type Volume Snapshot is supported"